modules/python/init.zsh

```diff
@@ -34,6 +34,48 @@ if (( ! $+commands[python] && ! $+commands[pyenv] )); then
   return 1
 fi
 
+function _python-workon-cwd {
+  # Check if this is a Git repo
+  local GIT_REPO_ROOT=""
+  local GIT_TOPLEVEL="$(git rev-parse --show-toplevel 2> /dev/null)"
+  if [[ $? == 0 ]]; then
+    GIT_REPO_ROOT="$GIT_TOPLEVEL"
+  fi
+  # Get absolute path, resolving symlinks
+  local PROJECT_ROOT="${PWD:A}"
+  while [[ "$PROJECT_ROOT" != "/" && ! -e "$PROJECT_ROOT/.venv" \
+            && ! -d "$PROJECT_ROOT/.git"  && "$PROJECT_ROOT" != "$GIT_REPO_ROOT" ]]; do
+    PROJECT_ROOT="${PROJECT_ROOT:h}"
+  done
+  if [[ "$PROJECT_ROOT" == "/" ]]; then
+    PROJECT_ROOT="."
+  fi
+  # Check for virtualenv name override
+  local ENV_NAME=""
+  if [[ -f "$PROJECT_ROOT/.venv" ]]; then
+    ENV_NAME="$(cat "$PROJECT_ROOT/.venv")"
+  elif [[ -f "$PROJECT_ROOT/.venv/bin/activate" ]];then
+    ENV_NAME="$PROJECT_ROOT/.venv"
+  elif [[ "$PROJECT_ROOT" != "." ]]; then
+    ENV_NAME="${PROJECT_ROOT:t}"
+  fi
+  if [[ -n $CD_VIRTUAL_ENV && -n $VIRTUAL_ENV ]]; then
```

Comment:
That would make sense... good catch.

It's worth noting that we might not need to call "deactivate" then "workon" if they don't match but your fix should still work.